src/backend/distributed/utils/listutils.c

```diff
@@ -165,10 +165,16 @@ StringJoin(List *stringList, char delimiter)
 	StringInfo joinedString = makeStringInfo();
 
 	const char *command = NULL;
+	int listSize = list_length(stringList);
+	int curIndex = 0;
 	foreach_ptr(command, stringList)
 	{
 		appendStringInfoString(joinedString, command);
-		appendStringInfoChar(joinedString, delimiter);
+		if (curIndex != (listSize - 1))
```

Comment:
Maybe prepend and use `curIndex > 0`? saves a variable

Reply:
good idea